src/test/compile-fail/closure-expected-type/expect-fn-supply-fn-multiple.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+type Different<'a, 'b> = &'a mut (&'a (), &'b ())
+type Same<'a> = Different<'a, 'a>;
+
+fn with_closure_expecting_different<F>(_: F)
+    where F: for<'a, 'b> FnOnce(Different<'a, 'b>)
+{
+}
+
+fn with_closure_expecting_different_anon<F>(_: F)
+    where F: FnOnce(Different<'_, '_>)
+{
+}
+
+fn supplying_different() {
```

Comment:
no error pattern? this shouldn't have errors in it.

Reply:
You should also have tests for all 4 ways of mixing Different/Same, to check that they have the same behavior (if the bound is `F: FnOnce(Different<'_, '_>)` but the closure is `|x: Same|`, that would be expected not to compile).

Reply:
The tests are just not passing yet. Will fix.